src/validation.ts

```diff
@@ -138,12 +139,7 @@ export function validateSettingsTCP(settings: TcpSettings): boolean {
 }
 
 function validateHttpTarget(target: string): boolean {
-  try {
-    const url = new URL(target);
-    return url.protocol === 'https:' || url.protocol === 'http:';
-  } catch (_) {
-    return false;
-  }
+  return Boolean(validUrl.isWebUri(target));
```

Comment:
I think we could also add a few extra test cases to `src/validation.test.ts`, just ensuring that what was false-positive until now (e.g. `target = "https://suraj/dev"`) is validated correctly in the future.

Reply:
Yeah... so I did that, and apparently https://suraj/dev isn't an invalid web uri according to spec. So this doesn't really fix that original case, but should catch other problems we weren't handling